test cases/common/28 try compile/meson.build

```diff
@@ -31,4 +34,15 @@ foreach lang : ['c', 'cpp']
   if compiler.compiles(files('invalid.c'), name : 'file should fail')
     error('Compiler ' + compiler.get_id() + ' returned true on broken code.')
   endif
+
+  # MSVC does not support #warning instruction
```

Comment:
I think we should add similar tests for links and runs and then this is good.

Reply:
Expanded tests to check the full matrix werror=true,false and compiles,links,runs.